src/coreclr/src/gc/gc.cpp

```diff
@@ -38718,19 +38500,22 @@ size_t GCHeap::ApproxTotalBytesInUse(BOOL small_heap_only)
 
     if (!small_heap_only)
     {
-        heap_segment* seg2 = generation_start_segment (pGenGCHeap->generation_of (max_generation+1));
-
-        while (seg2 != 0)
+        for (int i = uoh_start_generation; i < total_generation_count; i++)
         {
-            totsize += heap_segment_allocated (seg2) -
-                heap_segment_mem (seg2);
-            seg2 = heap_segment_next (seg2);
-        }
+            heap_segment* seg2 = generation_start_segment (pGenGCHeap->generation_of (i));
 
-        //discount the fragmentation
-        generation* loh_gen = pGenGCHeap->generation_of (max_generation+1);
-        size_t frag = generation_free_list_space (loh_gen) + generation_free_obj_space (loh_gen);
-        totsize -= frag;
+            while (seg2 != 0)
+            {
+                totsize += heap_segment_allocated (seg2) -
+                    heap_segment_mem (seg2);
+                seg2 = heap_segment_next (seg2);
+            }
+
+            //discount the fragmentation
+            generation* gen2 = pGenGCHeap->generation_of (i);
```

Comment:
nit - uoh_gen seems like a better name than gen2 (I'm aware that it says seg2 above but gen2 means something specific)